mpas_analysis/shared/mpas_xarray/mpas_xarray.py

```diff
@@ -141,59 +141,114 @@ def ensure_list(alist):  # {{{
+        1850-01-01
+
+        1850-01-01 00:00:00
 
-    The variable(s) pointed to by timestr should contain time information as a
-    date string, a floating-point number of days or a number of days
-    represented as a pandas timedelta (in ns).  The result is a list of
-    datetimes corresponding to the input dates offset as appropriate by the
-    yearoffset.
 
     Xylar Asay-Davis
-    Last modified: 12/05/2016
+
+    Last modified: 01/26/2017
     """
 
-    if isinstance(timestr, (tuple, list)):
+    calendar = '365_day'  # same as 'no_leap'
+
+    if isinstance(inTimeVarName, (tuple, list)):
         # we want to average the two
-        assert(len(timestr) == 2)
-        starts = get_datetimes(ds, timestr[0], yearoffset)
-        ends = get_datetimes(ds, timestr[1], yearoffset)
-        datetimes = [starts[i] + (ends[i] - starts[i])/2
-                     for i in range(len(starts))]
-        return datetimes
-
-    time_var = ds[timestr]
-
-    if time_var.dtype == '|S64':
-        # this is a variable like date strings like 'xtime'
-        time = [''.join(atime).strip() for atime in time_var.values]
-        datetimes = [datetime.datetime(yearoffset + int(x[:4]), int(x[5:7]),
-                                       int(x[8:10]), int(x[11:13]),
-                                       int(x[14:16]), int(x[17:19]))
-                     for x in time]
-    elif time_var.dtype == 'float64':
+        assert(len(inTimeVarName) == 2)
+        ds_start = parse_time(ds, inTimeVarName[0], inReferenceDate,
+                              outTimeVarName, outReferenceDate)
+        ds_end = parse_time(ds, inTimeVarName[1], inReferenceDate,
+                            outTimeVarName, outReferenceDate)
+        starts = ds_start[outTimeVarName].values
+        ends = ds_end[outTimeVarName].values
+        print 'averaging'
+        print starts, ends
+        print 'diff'
+        print [(ends[i] - starts[i])/2 for i in range(len(starts))]
+        # replace the time in starts with the mean of starts and ends
+        ds_start[outTimeVarName] = [starts[i] + (ends[i] - starts[i])/2
+                  for i in range(len(starts))]
+
+        print 'result'
+        print ds_start[outTimeVarName]
+        print 'done'
+        return ds_start
+
+    timeVar = ds[inTimeVarName]
+
+    if timeVar.dtype == '|S64':
+        # this is an array of date strings like 'xtime'
+        inCDFTime = netcdftime.utime(
+            'days since {}'.format(inReferenceDate),
+            calendar='365_day')
+        outCDFTime = netcdftime.utime(
+            'days since {}'.format(outReferenceDate),
+            calendar='365_day')
+        cfTimes = []
+        for xtime in timeVar.values:
+            # convert to string
+            timeString = ''.join(xtime).strip()
+            # convert to DatetimeNoLeap
+            date = netcdftime.DatetimeNoLeap(year=int(timeString[0:4]),
+                                             month=int(timeString[5:7]),
```

Comment:
It wouldn't need to.  There is a way of getting the appropriate `netcdftime.datetime` object for a specified calendar.  I just did this for our convenience and to make it explicit.

These comments are very helpful!